pkg/operator/resource_reconciler.go

```diff
@@ -666,13 +671,22 @@ func (rr *ResourceReconciler) processNextReconcileItem(ctx context.Context) bool
 	defer rr.reconcileQ.Done(key)
 	defer rr.statusQ.Add(key) // enqueues the object's key to update the status subresource
 
+	// Check if reconciliation should be delayed
```

Comment:
I wonder if we shouldn't move this earlier (e.g. right after `rr.reconcileQ.Get()`)?

Reply:
since Done() and Add() is called through `defer` I believe current approach is correct, correct me if I am wrong :)

Reply:
I expect that we don't enqueue to statusQ if the object isn't effectively reconciled.